### PR TITLE
[#10952] Submission links sent to instructors: Also give join link if not joined yet

### LINKED ISSUE #10952: Submission links sent to instructors: Also give join link if not joined yet
Scenario:
* A session has questions to be answered by instructors
* Instructors are sent emails containing a link to submit responses
* Unlike links sent to students, instructors need to join the course before they can submit responses

Suggestion:
* If the instructor has not joined yet, the email can also contain the join link with a note that they need to join before they can submit. If not, some might attempt to submit and contact us for tech support saying they receive an 'unauthorized user' error.

### PR DESCRIPTION
**Pull request by pPris**
Fixes #10952

**Outline of Solution**

The current feedback session email template has a placeholder called {instructorfragment}, which is only populated when the instructor is being sent a copy of the email that students receive.

I followed a similar method and created a variable that is always an empty string unless the instructor hasn't joined yet the course yet. In that case, the variable is assigned to a reminder template that's populated with the join link.

I'm not too sure of the words to use and where the reminder should be placed in the feedback session email template. (@damithc for your input perhaps?)

I'm also not very sure how to generate a populated version of the email template on the dev server side, is it possible to see how the final email would look like? (for src/main/resources/userEmailTemplate-feedbackSession.html)

### CODE REVIEW

File: src/main/resources/instructorEmailFragment-reminderToJoinCourseBeforeSubmittingResponse.html
Comment (damithc) [inline src/main/resources/instructorEmailFragment-reminderToJoinCourseBeforeSubmittingResponse.html:9]:
```suggestion
    Below is a copy of the original email sent to you earlier, which contains instructions on how to join the course.
```

File: src/main/resources/instructorEmailFragment-reminderToJoinCourseBeforeSubmittingResponse.html
Comment (damithc) [inline src/main/resources/instructorEmailFragment-reminderToJoinCourseBeforeSubmittingResponse.html:2]:
```suggestion
    Note that you will need to join the course as an instructor
```

**Review by damithc — COMMENTED** [review]
@pPris Added some comments regarding the wording. Note that there is a plan to allow instructors submit/view without having to join first, just like students. But this fix can be useful until that feature is implemented.

File: src/main/java/teammates/logic/api/EmailGenerator.java
Comment (daongochieu2810) [inline src/main/java/teammates/logic/api/EmailGenerator.java:793]: Yes we should abstract out repeated code, just rmb to remove this comment after u have done so

File: src/main/java/teammates/logic/api/EmailGenerator.java
Comment (daongochieu2810) [inline src/main/java/teammates/logic/api/EmailGenerator.java:403]: create a helper function for this chunk

File: src/main/java/teammates/logic/api/EmailGenerator.java
Comment (daongochieu2810) [inline src/main/java/teammates/logic/api/EmailGenerator.java:393]: I dont think combining `instructorJoinReminderFragment` with `additionalContactInformation` is a good idea, lets keep them separate

**Review by daongochieu2810 — CHANGES_REQUESTED** [review]
Changes look great! I left some comments

File: src/main/java/teammates/logic/api/EmailGenerator.java
Comment (pPris) [reply src/main/java/teammates/logic/api/EmailGenerator.java]: I've made the requested changes, there's an extra placeholder {additionalNotes} before {additionalContactInformation}

**Review by daongochieu2810 — APPROVED** [review]
Changes look good to me! Thanks for the contribution

File: src/main/resources/instructorEmailFragment-reminderToJoinCourseBeforeSubmittingResponse.html
Comment (rrtheonlyone) [inline src/main/resources/instructorEmailFragment-reminderToJoinCourseBeforeSubmittingResponse.html:11]: we can remove this comment, code duplication is understandable. The inviter information is not stored because there is no need to store it.

**Review by rrtheonlyone — APPROVED** [review]
LGTM 👍 Good to merge after you fix the conflicts @pPris and address minor comment below  

Email Output Looks good:

Screenshot 2021-07-25 at 7 03 26 PM

File: src/main/resources/instructorEmailFragment-reminderToJoinCourseBeforeSubmittingResponse.html
Comment (pPris) [reply src/main/resources/instructorEmailFragment-reminderToJoinCourseBeforeSubmittingResponse.html]: Hi @rrtheonlyone, I've fixed the conflicts and removed this comment. I realised I overlooked tests for this pr so I added a few lines into EmailGeneratorTest.java. please have a look before merging thank you!

**Pull request merged**
